### PR TITLE
Fix clear relations on modify issue

### PR DESCRIPTION
**Pull request by tomgross**
Details see https://github.com/plone/Products.CMFPlone/issues/2384

### CODE REVIEW

**Comment by tomgross** [comment]
@icemac @thet @pbauer mind to review?

File: src/z3c/relationfield/event.py [hunk 1]
```diff
@@ -73,7 +73,8 @@ def updateRelations(obj, event):
     # get errors involving buckets changing size
     rels = list(catalog.findRelations({'from_id': obj_id}))
     for rel in rels:
-        catalog.unindex(rel)
+        if hasattr(obj, rel.from_attribute):
+            catalog.unindex(rel)
```
Comment (mgedmin) [inline src/z3c/relationfield/event.py:77]: It would be very nice to have a unit test for this situation.

**Comment by tomgross** [comment]
@mgedmin Thanks for the review. I added a test demonstrating the change.

**Review by mgedmin — COMMENTED** [review]
It looks good to me, but I don't have the domain knowledge to understand the problem or the fix.

I'd say if nobody else steps up to review, feel free to just merge.

**Comment by ale-rt** [comment]
It also seems right to me

File: src/z3c/relationfield/event.py [hunk 2]
```diff
@@ -73,7 +73,8 @@ def updateRelations(obj, event):
     # get errors involving buckets changing size
     rels = list(catalog.findRelations({'from_id': obj_id}))
     for rel in rels:
-        catalog.unindex(rel)
+        if hasattr(obj, rel.from_attribute):
```
Comment (thet) [inline src/z3c/relationfield/event.py:76]: ``hasattr`` has the drawback to hide any error when accessing, but I think this is fine in this context as ``from_attribute`` isn't calculated.

**Comment by thet** [comment]
LGTM

**Review by thet — APPROVED** [review]
LGTM

**Comment by tomgross** [comment]
Since there are no objections, I'm merging this.

**Pull request merged**